### PR TITLE
Documentation for content media type and content encoding

### PR DESCRIPTION
**Pull request by wurzka**
With #231, we added annotations for content media type and content encoding.
This pull request adds the corresponding documentation.

### CODE REVIEW

**Comment by larshp** [comment]
but well, this is not a feature for ABAP file formats, right?

**Comment by wurzka** [comment]
Yes, this should not be used for AFF. Hence, we wanted to document it here instead of the AFF-repo

File: README.md [hunk 1]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
 The latest abapGit build can be downloaded [here](https://raw.githubusercontent.com/abapGit/build/main/zabapgit_standalone.prog.abap).
 
-The reports and their purpose
-* `z_generate_json_schema` generates the ABAP file formats JSON Schema (minimal requirement is v7.54)
-* `z_generate_repo` generates files and folder hierarchy for upload and maintain the [abap-file-formats](https://github.com/SAP/abap-file-formats) repository (requires ABAP file format framework)
-
 The development of the ABAP file formats tools takes place in the `Z` namespace.
 Coding policies are stated and verified by [abaplint](https://github.com/marketplace/abaplint), see configuration file [here](abaplint.jsonc), that run on every pull request.
 
+## Usage
```
[the remark this answers is not in the document]
Comment (schneidermic0) [reply README.md:24]:
```suggestion
## How To Use
```

File: README.md [hunk 2]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
+This repository mainly consists of two programs:
+* `z_generate_json_schema` generates JSON Schemas (minimal requirement is v7.54) from ABAP types
+* `z_generate_repo` generates files and folder hierarchy for upload and maintain the [abap-file-formats](https://github.com/SAP/abap-file-formats) repository (requires ABAP file format framework)
+
+To add more information to the JSON Schema than that provided by the ABAP type, ABAP Doc is parsed.
+Different possibilities on adding more information via annotations are listed [here](https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#writing-JSON-schema-with-ABAP-types).
+
+Additionally to the AFF-related annotation, one can also use the following:
```
Comment (schneidermic0) [inline README.md:32]:
```suggestion
Additionally to the ABAP-file-formats-related annotations, one can also use the following:
```

File: README.md [hunk 3]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
+### Content Media Type
+ To specify the content media type of the content of a string field, the annotation
+ ```abap
+ "! $contentMediaType 'mediaType'
+ ```
+ followed by the media type surrounded by single quotation marks is used.
+ This annotation can only be used for character like ABAP types, i.e., types that are mapped to JSON type `string`.
+
+ ### Content Encoding
```
Comment (schneidermic0) [inline README.md:41]: I think we should remove the space in the beginning:

```suggestion
### Content Encoding
```

**Comment by schneidermic0** [comment]
> but well, this is not a feature for ABAP file formats, right?

Any suggestions where we should document the behavior of the generator? As @wurzka wrote. Since it is not a feature of AFF, we moved the documentation to this repository.

**Comment by larshp** [comment]
I think the content-type feature would typically be needed in a REST setup, which is already covered else-ware

I suggest keeping the feature list as small as possible, as it will keep everything working

**Comment by schneidermic0** [comment]
> I suggest keeping the feature list as small as possible, as it will keep everything working

We wouldn't add if it was not needed. ;) It's really about transferring non-JSON data within a JSON document. See also https://json-schema.org/understanding-json-schema/reference/non_json_data.html

As mentioned before it is not related to the ABAP-file-formats. However, forking the tooling just for this use case seem also a bit over-engineered, isn't it?

**Comment by larshp** [comment]
true

but if you choose to share the use-case, then it will be possible to share alternative solutions

**Comment by schneidermic0** [comment]
> but if you choose to share the use-case, then it will be possible to share alternative solutions

I need to check whether I can already share it.

File: README.md [hunk 4]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
 The latest abapGit build can be downloaded [here](https://raw.githubusercontent.com/abapGit/build/main/zabapgit_standalone.prog.abap).
 
-The reports and their purpose
-* `z_generate_json_schema` generates the ABAP file formats JSON Schema (minimal requirement is v7.54)
-* `z_generate_repo` generates files and folder hierarchy for upload and maintain the [abap-file-formats](https://github.com/SAP/abap-file-formats) repository (requires ABAP file format framework)
-
 The development of the ABAP file formats tools takes place in the `Z` namespace.
 Coding policies are stated and verified by [abaplint](https://github.com/marketplace/abaplint), see configuration file [here](abaplint.jsonc), that run on every pull request.
 
+## How To Use
```
Comment (albertmink) [inline README.md:24]:
```suggestion
## How to Use
```

File: README.md [hunk 5]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
+This repository mainly consists of two programs:
+* `z_generate_json_schema` generates JSON Schemas (minimal requirement is v7.54) from ABAP types
+* `z_generate_repo` generates files and folder hierarchy for upload and maintain the [abap-file-formats](https://github.com/SAP/abap-file-formats) repository (requires ABAP file format framework)
```
Comment (albertmink) [inline README.md:27]:
```suggestion
* `z_generate_repo` generates files and folder hierarchy for uploading and maintaining the [abap-file-formats](https://github.com/SAP/abap-file-formats) repository (requires ABAP file format framework)
```

File: README.md [hunk 6]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
+
+To add more information to the JSON Schema than that provided by the ABAP type, ABAP Doc is parsed.
+Different possibilities on adding more information via annotations are listed [here](https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#writing-JSON-schema-with-ABAP-types).
```
Comment (albertmink) [inline README.md:30]:
```suggestion
Different possibilities for adding more information via annotations are listed [here](https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#writing-JSON-schema-with-ABAP-types).
```

File: README.md [hunk 7]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
+
+Additionally to the ABAP-file-formats-related annotations, one can also use the following:
+### Content Media Type
+ To specify the content media type of the content of a string field, the annotation
+ ```abap
+ "! $contentMediaType 'mediaType'
+ ```
+ followed by the media type surrounded by single quotation marks is used.
+ This annotation can only be used for character like ABAP types, i.e., types that are mapped to JSON type `string`.
```
Comment (albertmink) [inline README.md:39]:
```suggestion
 This annotation can only be used for characters like ABAP types, i.e., types that are mapped to JSON type `string`.
```

File: README.md [hunk 8]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
+
+### Content Encoding
+ The encoding used to store the content of a string field can be specifed with the annotation
```
Comment (albertmink) [inline README.md:42]:
```suggestion
 The encoding used to store the content of a string field can be specified with the annotation
```

File: README.md [hunk 9]
```diff
@@ -18,12 +18,34 @@ Detailed information including third-party components and their licensing/copyri
...
+ ```abap
+ "! $contentEncoding 'encoding'
+ ```
+ followed by the encoding surrounded by single quotation marks.
+ Possible values for the encoding are `7bit`, `8bit`, `binary`, `quoted-printable`, `base16`, `base32` and `base64`.
+ This annotation can also only be used for character like ABAP types, i.e., types that are mapped to JSON type `string`.
```
Comment (albertmink) [inline README.md:48]:
```suggestion
 This annotation can also only be used for characters like ABAP types, i.e., types that are mapped to JSON type `string`.
```

**Review by albertmink — CHANGES_REQUESTED** [review]
Spell checker suggestions

**Comment by wurzka** [comment]
Thanks @albertmink !

**Review by albertmink — COMMENTED** [review]
Introducing features that are not (yet?) relevant for ABAP file formats here is a potential risk, right? Considering there is not much of guidance/explanation for the intended usage and its prominent appearance on the readme.

---
Let's keep this in mind and observe it carefully. Approving and moving on 🚂

**Review by albertmink — APPROVED** [review]

**Comment by schneidermic0** [comment]
@wurzk, @larshp and I have just decided in a call to merge the changes for now.

**Pull request merged**